LibraBFT/Impl/Consensus/Types.agda

```diff
@@ -166,4 +233,17 @@ module LibraBFT.Impl.Consensus.Types where
     s : RoundManager → PendingVotes → RoundManager
     s rm pv = record rm { ₋rmEC = (₋rmEC rm) [ rmRoundState ∙ rsPendingVotes := pv ]  }
 
+  lSafetyRules : Lens RoundManager SafetyRules
+  lSafetyRules = mkLens' g s
+    where
+    g : RoundManager → SafetyRules
+    g rm = ₋rmEC rm ^∙ rmSafetyRules
+
+    s : RoundManager → SafetyRules → RoundManager
+    s rm sr = record rm { ₋rmEC = (₋rmEC rm) [ rmSafetyRules := sr ]}
+
+  lPersistentSafetyStorage : Lens RoundManager PersistentSafetyStorage
+  lPersistentSafetyStorage = lSafetyRules ∙ srPersistentStorage
 
+  lSafetyData : Lens RoundManager SafetyData
```

Comment:
This name (and others like it above, I think) causes a disconnect with the Haskell code, which uses `pssSafetyData` here, which is derived using `obmMakeClassy`.  Not sure if it makes sense to try to emulate `obmMakeClassy`.  More pragmatically, maybe we could just establish a naming convention, so for example, we might use `pssSafetyData-RoundManager` or `pssSafetyData-derived` or something so we alert the reader to why it's different in each case.  Thoughts?

Reply:
all the `l*` lens are direct analogues of 
```
instance RWSafetyRules (RoundManager a) where
  lSafetyRules = lens _rmSafetyRules (\x y -> x { _rmSafetyRules = y})
```
There is NO difference (except in the way the Agda defines `LBFT` compared to the haskell version.

Reply:
"pssSafety" != "lSafetyRules".  That's a difference.  I am saying this causes a disconnect for anyone comparing the Haskell code and the Agda model thereof.  As mentioned in my comment, I'm not suggesting we try to make them identical, but rather establish a convention to make it easier on the reader to navigate.

Reply:
But I see you have merged already, so let's keep this in mind for later.

Reply:
I do not think we are on the same page. 

`pssSafetyData` is a field inside `PersistentSafetyStorage`

`lSafetyRules` is a lens from `RoundManager` to `SafetyRules`

they are NOT the same type

Reply:
How _could_ we be on the same page, given that you're on the wrong one?  😝   Unless I am of course, but I double checked and I believe:

* `_pssSafetyData` (note underscore) is a field inside `PersistentSafetyStorage`
* `pssSafetyData` is a lens generated by `obmMakeClassy` (I think the instance you quoted above is part of how it is generated, right?)

Maybe I should be more explicit about the disconnect I'm commenting on.  In `constructAndSignVoteM.continue2`, we have in the Haskell `pssSafetyData .= safetyData` whereas in the Agda model .... ah, actually, I mean the _future_ Agda model that @cwjnkins is already fleshing out in another branch :smile: [SORRY!],  it says `lSafetyData ∙= safetyData1`, which looks different.  We are using _different_ lenses in the two contexts, even though they're equivalent (except that they have different type constraints in the Haskell code, which is about granularity of access capabilities.

Let's discuss at higher bandwidth next time we meet.

Reply:
Now I see - you are talking about a USE of `lSafetyData`.
In that case, the branch/location you point to would be better to do the update so it looks like the Haskell code.

Reply:
Yes, the USE in a future branch :smile:.  Sorry for being unclear and expecting you to be clairvoyant (I still expect it though :smile:).  My comments are regarding _how_ we can make it look like the Haskell code, given that the Haskell code is using a lens that is created by `obmMakeClassy` and we don't have an Agda counterpart for that.

Reply:
`obmMakeClassy` is not really the issue. 
That said, let's address this at the use site.